dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java

```diff
@@ -58,7 +66,57 @@ public Entry parseWorkflowContent(Entry entry, String filepath, String content,
 
     @Override
     public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, String primaryDescriptorFilePath) {
-        return new VersionTypeValidation(true, new HashMap<>());
+        if (minimalLanguageInterface instanceof RecommendedLanguageInterface) {
+            Optional<SourceFile> mainDescriptor = sourcefiles.stream()
+                    .filter((sourceFile -> Objects.equals(sourceFile.getPath(), primaryDescriptorFilePath))).findFirst();
+            String content = null;
```

Comment:
You could instead have `String content = mainDescriptor.orElseGet(null)`, or something like that.